src/item_factory.cpp

```diff
@@ -1744,11 +1757,76 @@ void Item_factory::load_pet_armor( const JsonObject &jo, const std::string &src
 
 void islot_armor::load( const JsonObject &jo )
 {
-    optional( jo, was_loaded, "encumbrance", encumber, 0 );
-    // Default max_encumbrance will be set to a reasonable value in
-    // finalize_post
-    optional( jo, was_loaded, "max_encumbrance", max_encumber, -1 );
-    optional( jo, was_loaded, "coverage", coverage, 0 );
+    if( jo.has_array( "coverage_data" ) ) {
+
+        for( const JsonObject &obj : jo.get_array( "coverage_data" ) ) {
+            random_armor_data tempData;
+            body_part_set temp_cover_data;
+            assign_coverage_from_json( obj, "parts", temp_cover_data, sided );
+            tempData.covers = temp_cover_data;
+
+            if( obj.has_array( "encumbrance" ) ) {
+                tempData.encumber = obj.get_array( "encumbrance" ).get_int( 0 );
+                tempData.max_encumber = obj.get_array( "encumbrance" ).get_int( 1 );
+            } else if( obj.has_int( "encumbrance" ) ) {
+                tempData.encumber = obj.get_int( "encumbrance" );
+                tempData.max_encumber = obj.get_int( "encumbrance" );
+            }
+            tempData.coverage = obj.get_int( "coverage" );
+
+            if( !data.empty() ) { // Uses copy-from
+                if( tempData.encumber ) {
+                    data[0].encumber = tempData.encumber;
+                }
+                if( tempData.max_encumber ) {
+                    data[0].max_encumber = tempData.max_encumber;
+                }
+                if( tempData.coverage ) {
+                    data[0].coverage = tempData.coverage;
+                }
```

Comment:
These values being non-zero isn't the same as them being specified.  This won't allow e.g. overriding coverage to zero in an item using `copy-from`.  Is it possible to use the standard JSON loading infrastructure code (`optional`, etc.) instead of rolling your own `copy-from` support, or did you already try that and run into difficulties?

Reply:
Oh yeah, I tried. optional really doesnt like copy-from, because its "already read" the data of the same name, so it doesnt read the value, and simply returns 0. There might be a better way to do it but this is the best I could come up with.